superdesk/celery_app/context_task.py

```diff
@@ -0,0 +1,96 @@
+import asyncio
+import werkzeug
+
+from celery import Task
+from asyncio import Future
+from typing import Any, Callable, Tuple, Dict
+
+from superdesk.logging import logger
+from superdesk.errors import SuperdeskError
+from superdesk.celery_app.serializer import CELERY_SERIALIZER_NAME
+
+
+class HybridAppContextTask(Task):
+    """
+    A task class that supports running both synchronous and asynchronous tasks within the Flask application context.
+    It handles exceptions specifically defined in `app_errors` and logs them.
+    """
+
+    abstract = True
+    serializer = CELERY_SERIALIZER_NAME
+    app_errors = (SuperdeskError, werkzeug.exceptions.InternalServerError)
+
+    def get_current_app(self):
+        """
+        Method that is intended to be overwritten so the module gets to use the right app
+        context
+        """
+        from superdesk.core import get_current_app
+
+        return get_current_app()
+
+    def __call__(self, *args: Any, **kwargs: Any) -> Any:
+        """
+        Executes the task function, determining if it should be run synchronously or asynchronously.
+
+        Args:
+            args: Positional arguments passed to the task function.
+            kwargs: Keyword arguments passed to the task function.
+        """
+        with self.get_current_app().app_context():
```

Comment:
This will need changes once we go to Quart, as the `app_context` in Quart is async. But should be good for now

Reply:
I added a TODO comment here so we don't forget :)